edb/common/parsing.py

```diff
@@ -109,17 +111,19 @@ def decorator(func: Any):
     return decorator
 
 
-class NontermMeta(type):
-    def __new__(mcls, name, bases, dct):
-        result = super().__new__(mcls, name, bases, dct)
 
-        if name == 'Nonterm' or name == 'ListNonterm':
-            return result
+class Nonterm(parsing.Nonterm):
 
-        if not result.__doc__:
-            result.__doc__ = '%nonterm'
+    def __init_subclass__(cls, is_internal=False, **kwargs):
+        super().__init_subclass__(**kwargs)
 
-        for name, attr in result.__dict__.items():
+        if is_internal:
```

Comment:
Let's add a docstring explaining that we read method names and use them to generate `%reduce` docstrings.

(Might as well clean things up a bit while you are here)